### ISSUE TITLE
Comment Depth is always 0

### ISSUE THREAD

**Issue by mi5ty** [timber/timber#1707]
**comment.twig**

```
<div class="blog-comment {{comment.comment_type}} comment-depth-{{comment.depth}}" id="comment-{{comment.ID}}">

  <div class="col-sm-12 comments animalo">
    <div class="comments_content">
      <h6 class="comment_user">{{ comment.author.name }}:</h6>
        {{comment.comment_content|wpautop}}
      <span class="comment_reply">{{ function('comment_reply_link', commentReplyArgs, comment.ID, post.ID, comment.depth) }}</span>
    </div>
    <span class="bb dt w-100 separator">&nbsp;</span>
  </div>

</div>

```
**single.twig**

```
{% for cmt in post.get_comments() %}
  {% include "layouts/comment.twig" with {comment:cmt} %}
  {% for child_cmt in cmt.children %}
    {% include "layouts/comment.twig" with {comment:child_cmt} %}
  {% endfor %}
{% endfor %}
```

**functions.php**

`$context["commentReplyArgs"] = array('reply_text' => "Reply", 'depth' => 1, 'max_depth' => 10);`

comment depth is always 0. What am I doing wrong?

**Comment by jake46a** [comment]
mi5ty,

I figure it may be time to start giving back to this plugin, which I now use extensively building templates with Bootstrap. I also remember about 7 days of struggling with this.  I pulled this from my project for that, was a while back, but here goes and excuse some of the variable names, old school here, lol.

**single.php**

```
$argscf = array(

        'id_form'           => 'commentform',
        'title_reply'       => __('leave a thought', 'pwasite'),
        'title_reply_to'    => __('leave a thought to %s', 'pwasite'),
        'cancel_reply_link' => __('cancel thought', 'pwasite'),
        'label_submit'      => __('post thought', 'pwasite'),
        'comment_field' =>  '<p><textarea placeholder="Start typing..." id="comment" class="form-control" name="comment" cols="45" rows="8" aria-required="true"></textarea></p>',
        'comment_notes_after' => '<p class="form-allowed-tags">' . __('You may use these <abbr title="HyperText Markup Language">HTML</abbr> tags and attributes:', 'wp-bootstrap-starter') . '</p><div class="alert alert-info">' . allowed_tags() . '</div>',
        'class_submit' => 'submit btn btn-primary'
      );	


.....
$context['argscf'] = $argscf;
```

**single.twig**

```
...
 <div id="comments" class="comments-area">
      {% if post.comments.count == 1 %}
        <h2 class="comments-title mt-5">Thought on
          {{post.title}}
        </h2>
      {% elseif post.comments.count >1 %}
        <h2 class="comments-title mt-5">Thoughts on
          {{post.title}}
        </h2>
      {% endif %}
      <ul class="list-unstyled comment-list">
        {% if post.comments %}
          {% for cmt in post.comments %}
            {% include "comment.twig" with {
              comment : cmt
            } %}
          {% endfor %}
        {% endif %}
      </ul>
      {% if post.comment_status == "closed" %}
        <p>
          comments for this post are closed
        </p>
      {% else %}
        {% include "comment-form.twig" %}
      {% endif %}
    </div>
...
```

**comment.twig**

```
{% if comment.children %}
<li id="comment-{{comment.ID}}" {{ fn('comment_class', 'parent' , comment.id)}} >
  {% else %}
<li id="comment-{{comment.ID}}" {{ fn('comment_class', class , comment.id)}} >
{% endif %}
  <article id="div-comment-{{comment.ID}}" class="comment-body media mb-4">
    <img class="pull-left p-2" src="{{cmt.avatar(36)}}"></a>
      <div class="media-body">
        <div class="media-body-wrap card">
            <div class="card-header">
              <h5 class="mt-0">
                <cite class="fn">{{comment.author.name}}</cite>
                <span class="says">thinks:</span>
              </h5>
              <div class="comment-meta">
                <a href="{{ fn ('get_comment_link', comment.id ) }}">
                  <time>{{ cmt.date }}&nbsp;{{ cmt.time }}<time>
                </a>
              </div>
            </div>
            <div class="comment-content card-block p-2">
              {{comment.comment_content|wpautop}}
            </div>
            <footer class="reply comment-reply card-footer">{{cmt.reply_link}}</footer>
          </div>
          {% for cmt in comment.children %}

          <ul class="list-unstyled comment-list">
  					{% include "comment-child.twig" with {comment:cmt} %}
          </ul>
  				{% endfor %}

        </div>
      </article>
</l1>
```
(The comment-child.twig was an exact copy of comment.twig, I was in a hurry but like your approach there.)

**comment-form.twig**

```
<div class="comment-form">
    <h3> add comment </h3>

    <form id="form" class="comment-form" method="post" action="{{ site.site_url~'/wp-comments-post.php' }}">
        {% if  user %}
            <input type="hidden" name="email" value="{{ user.email }}">
            <input type="hidden" name="author" value="{{ user.name }}">
            <input type="hidden" name="url" value="{{ user.link }}">
        {% else %}
            <div>
                <label for="email"> email </label>
                <input required name="email" type="text" id="email"/>

                <label for="author"> username </label>
                <input required name="author" type="text" id="author"/>

                <label for="url"> url </label>
                <input name="url" id="url" type="text"/>
            </div>
        {% endif %}

        <div>
            <label for="comment"> comment </label>
            <textarea placeholder="comment text" name="comment" cols="60" rows="3" id="comment"></textarea>
        </div>

        <input name="comment_post_ID" value="{{ post.id }}" id="comment_post_ID" type="hidden">
        <input name="comment_parent" id="comment_parent" value="{{ comment.ID|default('0') }}" type="hidden">

        <div>
            <button type="submit" name="Submit" class="btn"> send </button>
            <button type="reset"> cancel </button>
            <p> Your comment will be revised by the site if needed. </p>
        </div>
    </form>
</div>

```

Basically, what I did was to use the built in functions tied to the comment form in WordPress handle the depth.  Hope that helps, and also as you can see a lot of bootstrap also, lol.

**Comment by Levdbas** [comment]
Closing this stale question.
